pkg/testing/integration_test.go

```diff
@@ -123,7 +124,7 @@ func (s *IntegrationSuite) TestRun(c *C) {
 	c.Assert(err, IsNil)
 
 	// Set Controller namespace and service account
-	os.Setenv(kube.PodNSEnvVar, s.namespace)
+	os.Setenv(kube.PodNSEnvVar, consts.ControllerNS)
```

Comment:
Use of `os.Setenv` within separate goroutines is problematic.  It is likely safe now because the calls are all setting the same value. For the sake of maintainability it would be better to move these 2 lines to the `integrationSetup()` function in the other file.

Reply:
I have addressed this, but as @tdmanv  suggested, I have to make some improvements before be merge this.